apps/api/src/domain/models/song/commands/add-song.command.integration.spec.ts

```diff
@@ -0,0 +1,157 @@
+    payload: {
+        id: dummyUuid,
+        title: 'test-song-name (language)',
+        titleEnglish: 'test-song-name (English)',
+        contributorAndRoles: [],
+        lyrics: 'la la la',
+        audioURL: 'https://www.mysound.org/song.mp3',
+        lengthMilliseconds: 15340,
+    },
+};
+
+const validPayload = validCommandFSA.payload;
+describe('AddSong', () => {
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    let commandHandlerService: CommandHandlerService;
+
+    let arangoConnectionProvider: ArangoConnectionProvider;
+
+    beforeAll(async () => {
+        ({ testRepositoryProvider, commandHandlerService, arangoConnectionProvider } =
+            await setUpIntegrationTest({
+                ARANGO_DB_NAME: generateRandomTestDatabaseName(),
+            }));
+
+        commandHandlerService.registerHandler(
+            addSongCommandType,
+            new AddSongHandler(testRepositoryProvider)
+        );
+    });
+
+    afterAll(async () => {
+        await arangoConnectionProvider.dropDatabaseIfExists();
+    });
+
+    beforeEach(async () => {
+        await testRepositoryProvider.testSetup();
+    });
+
+    afterEach(async () => {
+        await testRepositoryProvider.testTeardown();
+    });
+
+    describe('when the payload is valid', () => {
+        it('should return ok', async () => {
+            await testRepositoryProvider.addFullSnapshot({
+                resources: {
+                    song: [],
+                },
+                tags: [],
+                connections: [],
+                categoryTree: [],
+            });
+
+            const result = await commandHandlerService.execute(validCommandFSA);
+
+            expect(result).toBe(Ack);
+        });
+    });
+
+    describe('when the payload has an invalid type', () => {
+        it('should return an error', async () => {
+            await testRepositoryProvider.addFullSnapshot(
+                buildInMemorySnapshot({
+                    resources: {
+                        song: [],
+                    },
+                })
+            );
+
+            const result = await commandHandlerService.execute({
+                ...validCommandFSA,
+                payload: {
+                    ...validPayload,
+                    id: [99],
+                },
+            });
+
+            expect(result).toEqual(
+                new InvalidCommandPayloadTypeError(addSongCommandType, [new InternalError('')])
+            );
+        });
+    });
+
+    describe('when the external state is invalid', () => {
+        describe('when there is already a song with the given ID', () => {
+            it('should return the expected error', async () => {
+                await testRepositoryProvider.addFullSnapshot(
+                    buildInMemorySnapshot({
+                        resources: {
+                            [ResourceType.song]: [
+                                getValidEntityInstaceForTest(ResourceType.song).clone({
```

Comment:
sp. - getValidEntityInstaceForTest

Reply:
good catch. Also - it should be `getValidResourceInstanceForTest`.  I'll fix this.

Reply:
done